apm-agent-plugins/apm-reactor-plugin/src/main/java/co/elastic/apm/agent/reactor/TracedSubscriber.java

```diff
@@ -236,8 +247,14 @@ static boolean isHookRegistered() {
                     return subscriber;
                 }
 
+                // use active span/transaction if directly active
                 AbstractSpan<?> active = tracer.getActive();
 
+                // fallback to using context-stored span/transaction if not already active
+                if (active == null) {
+                    active = subscriber.currentContext().getOrDefault(AbstractSpan.class, null);
+                }
+
```

Comment:
[for reviewer] this is what allows Reactor context propagation to retrieve the current context when it's not directly active, but has been stored, which happens when the wrapping occurs outside of any of the wrapped methods.